src/Widgets/IconGroupContainer.vala

```diff
@@ -65,6 +65,27 @@ namespace Gala
 			update_inserter_indices ();
 		}
 
+		public void remove_group_in_place (IconGroup group)
+		{
+			var thumb = (WorkspaceInsertThumb) group.get_previous_sibling ();
+			float thumb_width = thumb.get_width ();
+			float group_width = group.get_width ();
+			remove_group (group);
+			
+			x += thumb_width + group_width;
```

Comment:
Why does the x position of the container need to be moved when dragging begins?

Reply:
This is because if you start dragging the first icon group, the container will reallocate it's position and move to the left since there is no icon group there anymore. We need to add what was deducted back to `x` so that when you start dragging, the empty space still exists.